src/integration/url.ts

```diff
@@ -0,0 +1,27 @@
+const qs = new URLSearchParams(document.location.search || '')
+export const DEBUG_ANALYTICS = qs.has('DEBUG_ANALYTICS')
+export const ENV = qs.get('ENV')
+export const NETWORK = qs.get('NETWORK')
+export const RENDERER_TYPE = qs.get('ws') ? 'native' : 'web'
+export const PLATFORM = (navigator as any)?.userAgentData?.platform || navigator?.platform || 'unknown'
+export const HOSTNAME = document.location.hostname
+
+export function basedURL(url: string | URL, base: string | URL) {
```

Comment:
This name can be a bit confusing cause "base" is not really a verb in this case and the starting part of a URL usually is the origin. Maybe:

```
join(origin, url)
joinURLs(origin, url)

ensureOrigin(origin, url)

withOrigin(origin, url)
```